connector/src/main/java/org/geysermc/connector/network/translators/world/block/entity/PistonBlockEntity.java

```diff
@@ -0,0 +1,662 @@
+
+        Vector3d playerPos = Vector3d.from(collisionManager.getPlayerBoundingBox().getMiddleX(), collisionManager.getPlayerBoundingBox().getMiddleY() - (collisionManager.getPlayerBoundingBox().getSizeY() / 2), collisionManager.getPlayerBoundingBox().getMiddleZ());
+        Vector3d correctedPlayerPos = playerPos;
+
+        for (Object2IntMap.Entry<Vector3i> entry : attachedBlocks.object2IntEntrySet()) {
+            Vector3d blockPos = entry.getKey().toDouble().add(attachedBlockOffset);
+            correctedPlayerPos = handleBlockCollision(blockPos, entry.getIntValue(), correctedPlayerPos, playerCollision);
+        }
+        if (!correctedPlayerPos.equals(playerPos)) {
+            SessionPlayerEntity playerEntity = session.getPlayerEntity();
+            playerEntity.moveAbsolute(session, correctedPlayerPos.toFloat(), playerEntity.getRotation(), playerEntity.isOnGround(), false);
+            ClientPlayerPositionPacket playerPositionPacket = new ClientPlayerPositionPacket(playerEntity.isOnGround(), position.getX(), position.getY(), position.getZ());
+            session.sendDownstreamPacket(playerPositionPacket);
+        }
+    }
+
+    private Vector3d handleBlockCollision(Vector3d blockPos, int javaId, Vector3d playerPos, BoundingBox playerCollision) {
+        BlockCollision blockCollision = CollisionTranslator.getCollision(javaId, 0, 0, 0);
+        if (blockCollision != null) {
+            BoundingBox blockBoundingBox = blockCollision.getContainingBoundingBox();
+            if (blockBoundingBox != null && blockBoundingBox.checkIntersection(blockPos.getX(), blockPos.getY(), blockPos.getZ(), playerCollision)) {
+                // Slime blocks launch players away
+                // TODO setting the motion doesn't actually cause the bedrock player to move
+                String javaIdentifier = BlockTranslator.getJavaIdBlockMap().inverse().getOrDefault(javaId, "");
+                if (javaIdentifier.equals("minecraft:slime_block")) {
+                    SessionPlayerEntity playerEntity = session.getPlayerEntity();
+                    Vector3f motion = playerEntity.getMotion();
+                    double motionX = motion.getX();
+                    double motionY = motion.getY();
+                    double motionZ = motion.getZ();
+                    switch (orientation) {
+                        case DOWN:
+                        case UP:
+                            motionY = getMovement().getY();
+                            break;
+                        case NORTH:
+                        case SOUTH:
+                            motionZ = getMovement().getZ();
+                            break;
+                        case WEST:
+                        case EAST:
+                            motionX = getMovement().getX();
+                            break;
+                    }
+                    playerEntity.setMotion(Vector3f.from(motionX, motionY, motionZ));
```

Comment:
Do you ever update the player's motion to the server? Geyser doesn't do this automatically, you need to send a SetMotionPacket.